### PR TITLE
Add RHUI4 region validation for builds that require RHUI4

### PR DESCRIPTION
**Pull request by simratsatia**
Why is this PR required?
- As per the current functionality of validation, only RHUI3 regions can be tested during 'yum repolist', not RHUI4

What have we done?
- Renamed rhui-regions to rhui3-regions
- Added a new file for rhui4-regions
- Updated validate-vm-images.yaml to test RHUI3/RHUI4 regions based on parameters that are passed during running the ansible playbook

Pipeline link: https://msazure.visualstudio.com/One/_build/results?buildId=64418983&view=results

### CODE REVIEW

File: ansible_image_validation/validate-vm-images.yaml [hunk 1]
```diff
@@ -36,10 +38,16 @@
       include_tasks: validation-playbooks/per-vm-validation.yaml
       ignore_errors: yes
 
-    - name: Run per region validations
+    - name: Run per region validations for RHUI3
       include_tasks: validation-playbooks/per-region-validation.yaml
-      loop: "{{ rhui_regions }}"
-      when: item != "" and license_type == 'payg'
+      loop: "{{ rhui3_regions }}"
+      when: item != "" and license_type == 'payg' and build_on_rhui4 == 'false'
```
Comment (anujkmaurya) [inline ansible_image_validation/validate-vm-images.yaml:44]: let's add a check for not equal to byos images

File: ansible_image_validation/validate-vm-images.yaml [hunk 2]
```diff
@@ -36,10 +38,16 @@
...
+      ignore_errors: yes
+
+    - name: Run per region validations fpr RHUI4
+      include_tasks: validation-playbooks/per-region-validation.yaml
+      loop: "{{ rhui4_regions }}"
+      when: item != "" and license_type == 'payg' and build_on_rhui4 == 'true'
```
Comment (anujkmaurya) [inline ansible_image_validation/validate-vm-images.yaml:50]: same comment as above 
let's add a check for not equal to byos images

**Review by anujkmaurya — CHANGES_REQUESTED** [review]
added comments

File: ansible_image_validation/validate-vm-images.yaml [hunk 3]
```diff
@@ -17,9 +17,11 @@
         that:
           - "offer_type is defined"
           - "rhel_version is defined"
+          - "build_on_rhui4 is defined"
```
Comment (anujkmaurya) [inline ansible_image_validation/validate-vm-images.yaml:20]: where is the corresponding pr for passing this argument to this playbook?

**Review by anujkmaurya — CHANGES_REQUESTED** [review]

File: ansible_image_validation/validate-vm-images.yaml (hunk 3 above)
Comment (simratsatia) [reply ansible_image_validation/validate-vm-images.yaml:20]: https://msazure.visualstudio.com/One/_git/Compute-AzLinux-RedHat-Image/pullrequest/7155504

**Review by anujkmaurya — APPROVED** [review]

**Pull request merged**